### ISSUE TITLE
About the reproduced performance of LeViT?

### ISSUE THREAD

**Issue by Openning07** [lucidrains/vit-pytorch#94]
Hello guys.
Have you successfully reproduced the performance of LeViT?
I've attempted to train the LeViT, which seems to be among the top-performing ViT models. But I failed to reproduced the results, which are claimed in the paper of LeViT.

**Comment by EelcoHoogendoorn** [comment]
Just looking at the levit implementation; one apparent difference from the paper I noticed relates to this 'Following [54], each
batch normalization weight parameter that joins up with a residual connection is initialized to zero.'

Not seeing that reflected in the implementation; not sure if its likely to explain much of your observed difference though; though it might have a substantial impact?

**Comment by lucidrains** [comment]
I decided to add what @EelcoHoogendoorn noticed in the paper - @Openning07  do you want to give it a try? https://github.com/lucidrains/vit-pytorch/commit/0df15056624b18d7894adc4937021399d5dc027e

**Comment by Openning07** [comment]
I've tried it and it still cannot works.
Specifically,
when I used 5e-4 as the initial learning-rate, LeViT-128S came up with *nan loss* at Epoch 6 (I ran 3 times and the *nan loss* can be reproduced);
when I used 5e-5 as the initial learning-rate, the *nan loss* did not appear, but the accuracy seems to be much lower compared with DeiT-tiny (e.g., the performance drops about 10% at Epoch 120).

**Comment by btgraham** [comment]
LeViT is open sourced here: https://github.com/facebookresearch/LeViT
